go_lib/dependency/cr/cr.go

```diff
@@ -117,8 +117,7 @@ func (r *client) Image(ctx context.Context, tag string) (v1.Image, error) {
 
 	if r.options.timeout > 0 {
 		// add default timeout to prevent endless request on a huge image
-		ctxWTO, cancel := context.WithTimeout(ctx, r.options.timeout)
-		defer cancel()
+		ctxWTO, _ := context.WithTimeout(ctx, r.options.timeout) // nolint:govet
```

Comment:
```suggestion
        // Warning!: don't use cancel() in the defer func here. Otherwise *v1.Image outside this function would be inaccessible due to cancelled context, while reading layers, for example.
		ctxWTO, _ := context.WithTimeout(ctx, r.options.timeout) // nolint:govet
```